examples/md-flexible/YamlParser.cpp

```diff
@@ -0,0 +1,931 @@
+/**
+ * @file YamlParser.cpp
+ * @author N. Fottner
+ * @date 15/7/19
+ */
+
+#include "YamlParser.h"
+
+bool YamlParser::parseInput(int argc, char **argv) {
+  using namespace std;
+  bool displayHelp = false;
+  int option, option_index;
+  static struct option long_options[] = {{"filename", required_argument, nullptr, 'Y'},
```

Comment:
I'm not conviced that it is a good idea to merge the YAML and Command line parser in the same class. This seems a bit messy. However, it would still be nice if the CLI parser can overwrite the YAML one. I'll make a PR for this.

Reply:
done with #362